SYCL/Plugin/level_zero_sub_sub_device.cpp

```diff
@@ -25,89 +25,65 @@ using namespace std::chrono;
 #define INTER_NUM (150)
 #define KERNEL_NUM (2000)
 
-void run(std::vector<queue> &queues) {
+void make_queue_and_run_workload(std::vector<device> &subdevices) {
+
+  std::cout << "[important] create " << subdevices.size()
+            << " sycl queues, one for each sub-sub device" << std::endl;
+
   auto N = 1024 * 16;
   size_t global_range = 1024;
   size_t local_range = 16;
 
-  float *buffer_host0 = malloc_host<float>(N, queues[0]);
-  float *buffer_device0 = malloc_device<float>(N, queues[0]);
-
-  float *buffer_host1 = malloc_host<float>(N, queues[1]);
-  float *buffer_device1 = malloc_device<float>(N, queues[1]);
-
-  float *buffer_host2 = malloc_host<float>(N, queues[2]);
-  float *buffer_device2 = malloc_device<float>(N, queues[2]);
+  std::vector<queue> queues;
+  std::vector<float *> buffer_hosts;
+  std::vector<float *> buffer_devices;
+
+  // Create queues for each subdevice
+  for (auto subdevice : subdevices) {
+    queue temp(subdevice, {property::queue::enable_profiling(),
+                           property::queue::in_order()});
+    float *buffer_host = malloc_host<float>(N, temp);
+    float *buffer_device = malloc_device<float>(N, temp);
```

Comment:
The type is obvious from the template param, no need to spell it twice.

```suggestion
    auto *buffer_host = malloc_host<float>(N, temp);
    auto *buffer_device = malloc_device<float>(N, temp);
```